Twisted WebSocket test build failure

Unexpected build failure with no change in our code:

```Shell
================================================================================== FAILURES ==================================================================================
______________________________________________________ TestProdServerToConsClientProtocols.test_url_encoded_components _______________________________________________________

self = <test_twisted_websocket.TestProdServerToConsClientProtocols testMethod=test_url_encoded_components>

    def test_url_encoded_components(self):
        # This test is about getting percent encoded characters work in sequenceId or hostname
        sequence_id = u'sequence/ünicödé?/Name'
        self._create_server(url='ws://localhost:9006', producer=self.prod)
        self._create_client(
            url='ws://localhost:9006/sequence%2F%C3%BCnic%C3%B6d%C3%A9%3F%2FName/subscribe',
            consumer=self.cons
        )
    
        self.cproto.consumer = self.cons
    
>       self._connect()

ebu_tt_live/twisted/test/test_twisted_websocket.py:214: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ebu_tt_live/twisted/test/test_twisted_websocket.py:47: in _connect
    self.assertEquals(self.sproto.state, self.sproto.STATE_OPEN)
venv/lib/python2.7/site-packages/twisted/trial/_synctest.py:432: in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
E   FailTest: 0 != 3
-------------------------------------------------------------------------------- Captured log --------------------------------------------------------------------------------
websocket.py               227 ERROR    one or more reserved delimiters /?# present in path segment: u'sequence/\xfcnic\xf6d\xe9?/Name'
websocket.py               352 ERROR    one or more reserved delimiters /?# present in path segment: u'sequence/\xfcnic\xf6d\xe9?/Name'
```

Presumably this has been induced by a change to a dependency.

--- a/ebu_tt_live/twisted/websocket.py
+++ b/ebu_tt_live/twisted/websocket.py
@@ -3,7 +3,7 @@ from autobahn.twisted.websocket import WebSocketClientProtocol, WebSocketServerF
     listenWS, WebSocketClientFactory, connectWS
 
 from twisted.internet import interfaces, reactor
-from twisted.python import url as twisted_url
+from hyperlink import URL
 from zope.interface import implementer
 from logging import getLogger
 import json
@@ -16,7 +16,6 @@ from .base import IBroadcaster
 
 
 log = getLogger(__name__)
-
 
 class UserInputServerProtocol(WebSocketServerProtocol):
     def onOpen(self):
@@ -102,7 +101,7 @@ class EBUWebsocketProtocolMixin(object):
     def _parse_path(self, full_url):
         if not isinstance(full_url, six.text_type):
             full_url = six.text_type(full_url)
-        result = twisted_url.URL.fromText(full_url).asIRI()
+        result = URL.fromText(full_url).to_iri()
         sequence_identifier, action = result.path
         return sequence_identifier, action
 

--- a/setup.py
+++ b/setup.py
@@ -35,7 +35,8 @@ setup(
         "nltk",
         "sortedcontainers",
         "configman",
-        "six"
+        "six",
+        "hyperlink<17.2.0"  # This should be removed if https://github.com/python-hyper/hyperlink/issues/16 is fixed
     ],
     license="BSD3",
     packages=packages,

